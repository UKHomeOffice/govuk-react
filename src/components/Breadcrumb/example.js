import React from 'react';
import Breadcrumb from '.';

export default () => (
  <Breadcrumb>
    <a href="/section">Section 1</a>
  </Breadcrumb>
);
