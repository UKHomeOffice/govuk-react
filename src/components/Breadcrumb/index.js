// https://github.com/alphagov/govuk-frontend/blob/master/src/components/breadcrumb/_breadcrumb.scss
import React from "react";
import glamorous from "glamorous";
import PropTypes from "prop-types";

import { BLACK, GREY_1, YELLOW } from "govuk-colours";
import { MEDIA_QUERIES } from "../../constants/index";

const BreadcrumbContainer = glamorous.div({
  fontFamily: "nta, Arial, sans-serif",
  fontSize: "14px",
  lineHeight: "16px",
  marginTop: "15px",
  marginBottom: "10px",
  WebkitFontSmoothing: "antialiased",
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: "16px",
    lineHeight: "20px"
  }
});

const BreadcrumbList = glamorous.ul({
  margin: 0,
  padding: 0,
  listStyleType: "none",
  display: "block"
});

const BreadcrumbListItem = glamorous.li({
  position: "relative",
  display: "inline-block",
  whiteSpace: "no-wrap",
  marginBottom: "5px",
  marginLeft: "10px",
  paddingLeft: "15px",
  ":first-child": {
    marginLeft: 0,
    paddingLeft: 0,
    "::before": {
      display: "none"
    }
  },
  "::before": {
    content: " ",
    display: "block",
    position: "absolute",
    top: "-1px",
    bottom: "1px",
    left: "-3.31px",
    width: "7px",
    height: "7px",
    margin: "auto 0",
    transform: "rotate(45deg)",
    border: "solid",
    borderWidth: "1px 1px 0 0",
    borderColor: `${GREY_1}`
  },
  "> a": {
    color: `${BLACK}`,
    textDecoration: "underline",
    ":focus": {
      backgroundColor: `${YELLOW}`,
      outline: `3px solid ${YELLOW}`
    }
  }
});

const Breadcrumb = ({ children }) => (
  <BreadcrumbContainer>
    <BreadcrumbList>
      {children.length && children.map ? (
        children.map((child, i) => (
          <BreadcrumbListItem key={child.key || i}>{child}</BreadcrumbListItem>
        ))
      ) : (
        <BreadcrumbListItem>{children}</BreadcrumbListItem>
      )}
    </BreadcrumbList>
  </BreadcrumbContainer>
);

Breadcrumb.propTypes = {
  children: PropTypes.node.isRequired
};

export default Breadcrumb;
