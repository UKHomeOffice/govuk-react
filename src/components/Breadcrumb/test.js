import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount } from "enzyme";

import Breadcrumb from "./";

describe("breadcrumb", () => {
  const example = "example";
  const wrapper = <Breadcrumb>{example}</Breadcrumb>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("should render an unordered list", () => {
    const output = shallow(wrapper);
    expect(output.find("ul")).toBeTruthy();
    expect(output.find("ul > li")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
  });
});
