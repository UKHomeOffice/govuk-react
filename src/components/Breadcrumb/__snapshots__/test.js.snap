// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
<Breadcrumb>
  <glamorous(div)>
    <div
      className="css-v85tod"
    >
      <glamorous(ul)>
        <ul
          className="css-j76yvj"
        >
          <glamorous(li)>
            <li
              className="css-uxdkvh"
            >
              example
            </li>
          </glamorous(li)>
        </ul>
      </glamorous(ul)>
    </div>
  </glamorous(div)>
</Breadcrumb>
`;

exports[`enzyme.render 1`] = `
<div
  class="css-v85tod"
>
  <ul
    class="css-j76yvj"
  >
    <li
      class="css-uxdkvh"
    >
      example
    </li>
  </ul>
</div>
`;

exports[`enzyme.shallow 1`] = `
<glamorous(div)>
  <glamorous(ul)>
    <glamorous(li)>
      example
    </glamorous(li)>
  </glamorous(ul)>
</glamorous(div)>
`;
